src/CMakeLists.txt

```diff
@@ -70,14 +70,14 @@ include_directories(
     ${PROJECT_SOURCE_DIR}/src/benchmarklib/
     ${PROJECT_SOURCE_DIR}/src/lib/
     ${PROJECT_SOURCE_DIR}/src/plugins/
-
-    ${Boost_INCLUDE_DIRS}
 )
 
 # Include these libraries as system libraries to silence some warnings caused by -Weverything
 include_directories(
     SYSTEM
+    ${Boost_INCLUDE_DIRS}
     ${TBB_INCLUDE_DIR}
+    ${SQLITE3_INCLUDE_DIR}
```

Comment:
With my limited knowledge about CMake (mostly based on https://cliutils.gitlab.io/modern-cmake/), I think we should avoid adding include directories of third-party libraries with global functions such as `include_directories`. #2032 also mentions this. Instead, I suggest creating a library target with the include directories and linking the library target to another target. If the library target only contains header files, which would be true for the boost library here, this target can be an INTERFACE target.

I'm cleaning up some CMake files in #2421. In that PR, I also created a tbb library target, which can be linked to another library/executable then. This is a way we could also do it for boost and sqlite.

Reply:
I don't have a clue what you're talking about. :D 
But do you think #2421 will make it into the master soon? Merging these two lines would help me right now while working within the DE lab. If it's in the end about rewriting this include lines, this PR here wouldn't interfere with yours right?

Reply:
#2421 won't make it into the master soon. One CI stage fails and I have to debug it...

This PR would not interfere with #2421, but it also does not add include directories of third-party libraries properly IMO. I defined library targets for tbb, sqlite3, and boost in #2422 and linked them as system libraries. Would that also solve your issue? However, the CI run is still running, so we will see whether it succeeds.

Reply:
Never mind, #2422 also fails, and it will probably take some time to find the root cause. I'm okay with applying these changes for now and finding a cleaner later on.

Reply:
I would usually say "I'll have a look and help you", but I have absolutely  no idea about that whole cmake thing. :-/